### PR TITLE
Fixes #35147 - Hosts jobs card to show host job status

### PR DESCRIPTION
**Pull request by MariaAga**
* also changed the links in the dropdown to filter the hosts with `target_host_id` and not `host` for more correct results

### CODE REVIEW

File: app/controllers/job_invocations_controller.rb [hunk 1]
```diff
@@ -130,6 +130,24 @@ def cancel
     redirect_back(:fallback_location => job_invocation_path(@job_invocation))
   end
 
+  def preview_job_invocations_per_host
+    job_invocations = resource_base.search_for("targeted_host_id = #{params[:host_id]} and (status=#{params[:status]})").limit(params[:limit] || 3)
```
Comment (jeremylenz) [inline app/controllers/job_invocations_controller.rb:134]: I'm getting an error when clicking a link in the dropdown:

```
Invalid search query: Value 'rhel8b.fedora.example.com' is not valid for field 'targeted_host_id'
```

**Review by jeremylenz — COMMENTED** [review]
Tested with 2 hosts, one running and one not running.

:heavy_check_mark: Overall job status shows 50% completed, 50% failed
:heavy_check_mark: Recent jobs card shows succeeded on the host that succeeded
:heavy_check_mark: Recent jobs card shows failed on the host that failed

But the links are not working.

File: app/controllers/job_invocations_controller.rb [hunk 2]
```diff
@@ -130,6 +130,24 @@ def cancel
...
+    job_invocations = job_invocations.map do |job|
+      @job_invocation = job
+      template_invocation = job.template_invocations.find { |template_inv| template_inv.host_id == params[:host_id].to_i }
+      task = template_invocation.try(:run_host_job_task)
+      {
+        start_at: job.start_at,
+        description: job.description,
+        id: job.id,
+        status: task ? HostStatus::ExecutionStatus::ExecutionTaskStatusMapper.new(task).status : job.status,
+        status_label: task ? HostStatus::ExecutionStatus::ExecutionTaskStatusMapper.new(task).status_label : job.status_label,
```
Comment (adamruzicka) [inline app/controllers/job_invocations_controller.rb:145]: could we create the mapper only once?

**Review by jeremylenz — APPROVED** [review]
Thanks @MariaAga, the links work now!

Ack :+1:

**Comment by adamruzicka** [comment]
Thank you @MariaAga & @jeremylenz !

**Pull request merged**